From 145bc2ec334d05ef7d0d389ac26646196e24ae42 Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 11:59:03 -0400
Subject: [infiniband] core: Add RDMA_TRANSPORT_USNIC_UDP

Message-id: <09ff9d07a2bd79da7c3ba09b1d0a8cc900b63d0a.1410425016.git.avadai@redhat.com>
Patchwork-id: 93572
O-Subject: [Patch RHEL7.1 V1 007/197] IB/core: Add RDMA_TRANSPORT_USNIC_UDP
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Upinder Malhi <umalhi@cisco.com>

Add RDMA_TRANSPORT_USNIC_UDP which will be used by usNIC.

Signed-off-by: Upinder Malhi <umalhi@cisco.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 248567f79304b953ea492fb92ade097b62ed09b2)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/core/verbs.c b/drivers/infiniband/core/verbs.c
index b3acfc0..c1cfeda 100644
--- a/drivers/infiniband/core/verbs.c
+++ b/drivers/infiniband/core/verbs.c
@@ -118,7 +118,7 @@ rdma_node_get_transport(enum rdma_node_type node_type)
  case RDMA_NODE_RNIC:
   return RDMA_TRANSPORT_IWARP;
  case RDMA_NODE_USNIC:
-  return RDMA_TRANSPORT_USNIC;
+  return RDMA_TRANSPORT_USNIC_UDP;
  default:
   BUG();
   return 0;
@@ -136,6 +136,7 @@ enum rdma_link_layer rdma_port_get_link_layer(struct ib_device *device, u8 port_
   return IB_LINK_LAYER_INFINIBAND;
  case RDMA_TRANSPORT_IWARP:
  case RDMA_TRANSPORT_USNIC:
+ case RDMA_TRANSPORT_USNIC_UDP:
   return IB_LINK_LAYER_ETHERNET;
  default:
   return IB_LINK_LAYER_UNSPECIFIED;
diff --git a/include/rdma/ib_verbs.h b/include/rdma/ib_verbs.h
index facb5cd..e646ff8 100644
--- a/include/rdma/ib_verbs.h
+++ b/include/rdma/ib_verbs.h
@@ -75,7 +75,8 @@ enum rdma_node_type {
 enum rdma_transport_type {
  RDMA_TRANSPORT_IB,
  RDMA_TRANSPORT_IWARP,
- RDMA_TRANSPORT_USNIC
+ RDMA_TRANSPORT_USNIC,
+ RDMA_TRANSPORT_USNIC_UDP
 };
 
 enum rdma_transport_type
-- 
1.7.1